From: Steven Chamberlain <steven@pyro.eu.org>
Date: Fri, 27 Sep 2024 23:19:10 +0200
Subject: Define _GNU_SOURCE to use X/Open features

Define _GNU_SOURCE to get nftw() macros (XOPEN_SOURCE_EXTENDED)
strptime() (XOPEN_SOURCE), pthread_lock_t (POSIX.1-2001) and others.

Bug-Debian: https://bugs.debian.org/820155
---
 gw/dlr_spool.c             | 2 ++
 gw/smsc/http/clickatell.c  | 2 ++
 gw/smsc/smsc_http.c        | 2 ++
 test/test_file_traversal.c | 2 ++
 4 files changed, 8 insertions(+)

diff --git a/gw/dlr_spool.c b/gw/dlr_spool.c
index 2eba224..0085f65 100644
--- a/gw/dlr_spool.c
+++ b/gw/dlr_spool.c
@@ -64,6 +64,8 @@
  * Stipe Tolj <stolj at kannel dot org>
  */
 
+#define _GNU_SOURCE
+
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
diff --git a/gw/smsc/http/clickatell.c b/gw/smsc/http/clickatell.c
index d950e99..24cd9a6 100644
--- a/gw/smsc/http/clickatell.c
+++ b/gw/smsc/http/clickatell.c
@@ -61,6 +61,8 @@
  * Stipe Tolj <st@tolj.org>, <stolj@kannel.org>
  */
 
+#define _GNU_SOURCE
+
 #include "gwlib/gwlib.h"
 
 
diff --git a/gw/smsc/smsc_http.c b/gw/smsc/smsc_http.c
index 6929146..528395d 100644
--- a/gw/smsc/smsc_http.c
+++ b/gw/smsc/smsc_http.c
@@ -111,6 +111,8 @@
  * Tobias Weber <weber@wapme.de>
  */
 
+#define _GNU_SOURCE
+
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <unistd.h>
diff --git a/test/test_file_traversal.c b/test/test_file_traversal.c
index 013564e..4ae0850 100644
--- a/test/test_file_traversal.c
+++ b/test/test_file_traversal.c
@@ -58,6 +58,8 @@
  * test_file_traversal.c - simple file traversal testing
  */
 
+#define _GNU_SOURCE
+
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
